api/v1alpha1/addresspool_webhook.go

```diff
@@ -0,0 +1,168 @@
+
+	return addressPool.validateAddressPool(true)
+}
+
+// ValidateUpdate implements webhook.Validator so a webhook will be registered for AddressPool.
+func (addressPool *AddressPool) ValidateUpdate(old runtime.Object) error {
+	addresspoollog.Info("validate AddressPool update", "name", addressPool.Name)
+
+	return addressPool.validateAddressPool(false)
+}
+
+// ValidateDelete implements webhook.Validator so a webhook will be registered for AddressPool.
+func (addressPool *AddressPool) ValidateDelete() error {
+	addresspoollog.Info("validate AddressPool deletion", "name", addressPool.Name)
+
+	return nil
+}
+
+func cidrsOverlap(a, b *net.IPNet) bool {
+	return cidrContainsCIDR(a, b) || cidrContainsCIDR(b, a)
+}
+
+func cidrContainsCIDR(outer, inner *net.IPNet) bool {
+	ol, _ := outer.Mask.Size()
+	il, _ := inner.Mask.Size()
+	if ol == il && outer.IP.Equal(inner.IP) {
+		return true
+	}
+	if ol < il && outer.Contains(inner.IP) {
+		return true
+	}
+	return false
+}
+
+func parseCIDR(cidr string) ([]*net.IPNet, error) {
+	if !strings.Contains(cidr, "-") {
+		_, n, err := net.ParseCIDR(cidr)
+		if err != nil {
+			return nil, fmt.Errorf("invalid CIDR %q", cidr)
+		}
+		return []*net.IPNet{n}, nil
+	}
+
+	fs := strings.SplitN(cidr, "-", 2)
+	if len(fs) != 2 {
+		return nil, fmt.Errorf("invalid IP range %q", cidr)
+	}
+	start := net.ParseIP(strings.TrimSpace(fs[0]))
+	if start == nil {
+		return nil, fmt.Errorf("invalid IP range %q: invalid start IP %q", cidr, fs[0])
+	}
+	end := net.ParseIP(strings.TrimSpace(fs[1]))
+	if end == nil {
+		return nil, fmt.Errorf("invalid IP range %q: invalid end IP %q", cidr, fs[1])
+	}
+
+	var ret []*net.IPNet
+	for _, pfx := range ipaddr.Summarize(start, end) {
+		n := &net.IPNet{
+			IP:   pfx.IP,
+			Mask: pfx.Mask,
+		}
+		ret = append(ret, n)
+	}
+	return ret, nil
+}
+
+func parseAddresses(addressPool *AddressPool) ([]*net.IPNet, error) {
+	var CIDRs []*net.IPNet
+	for _, cidr := range addressPool.Spec.Addresses {
+		nets, err := parseCIDR(cidr)
+		if err != nil {
+			return nil, fmt.Errorf("invalid CIDR %q in pool %q: %s", cidr, addressPool.Name, err)
+		}
+		CIDRs = append(CIDRs, nets...)
+	}
+	return CIDRs, nil
+}
+
+func (addressPool *AddressPool) validateAddressPool(isNewAddressPool bool) error {
+	addressPoolList := &AddressPoolList{}
+	err := c.List(context.Background(), addressPoolList)
+	if err != nil {
+		return fmt.Errorf(fmt.Sprintf("Failed to get existing addresspool objects %s", err))
+	}
+
+	addressPoolCIDRS, err := parseAddresses(addressPool)
+	if err != nil {
+		return fmt.Errorf(fmt.Sprintf("Failed to parce addresses for %s %s", addressPool.Name, err))
+	}
+
+	for _, existingAddressPool := range addressPoolList.Items {
+		if existingAddressPool.Name == addressPool.Name {
+			// Check that the pool isn't already defined.
+			// Avoid errors when comparing the AddressPool to itself.
+			if isNewAddressPool {
+				return fmt.Errorf("duplicate definition of pool %q", addressPool.Name)
```

Comment:
Is this even possible?

Reply:
yes. but are you asking about a new addresspool or an existing one?